### PR TITLE
6655 sync api el offline

### LINKED ISSUE #6655: VC failover when EL is offline
VC should failover when the EL connected to a BN is offline

**Background** 

Question from user on Discord (obfuscate):

`Hi there, I am currently experimenting with fail over scenarios in my system. I have the Teku Beacon Node and Validator clients split out into separate processes. I have a primary BN connected to Besu and a secondary BN connected to Nethermind. I have configured the two BN endpoints in the VCs to examine the fail over behaviour. When I bring the primart BN down it fails over to the secondary BN almost instantly which is great! However, when I bring my primary EL client (Besu) down, the VC doesn't seem to failing over the secondary BN (with the healthy Nethermind node connected it).`

https://discord.com/channels/697535391594446898/697539289042649190/1058317129805484082

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description
Add an `el_offline` field to the `/eth/v1/node/syncing syncing` API response:
- Made the `SyncStateTracker` listen to the `ExecutionClientEventsChannel` to track the EL availability status
- Added a new `EL_OFFLINE` to the `SyncState`
- Added an acceptance test

## Fixed Issue(s)

This is an implementation of https://github.com/ethereum/beacon-APIs/pull/290
And will be used to fix https://github.com/ConsenSys/teku/issues/6655
## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/SyncingStatusAcceptanceTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    beaconNode.waitForEpochAtOrAbove(1);
+
+    beaconNode.expectNodeNotSyncing();
+
+    eth1Node.stop();
+
+    beaconNode.waitForLogMessageContaining("Make sure the Execution Client is online");
+
+    beaconNode.expectElOffline();
```
Comment (zilm13) [inline acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/SyncingStatusAcceptanceTest.java:76]: I'd add a check that it's not offline before we stopped execution client

**Review by zilm13 — APPROVED** [review]
LGTM
Just when we start with EL offline we are responding with `el_offline` `false` in first 30 seconds until we get error, we could change this behavior but it will mean that response is not correct when el is online on start, so it's a normal trade-off I think

File: acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/SyncingStatusAcceptanceTest.java (hunk 1 above)
Comment (mehdi-aouadi) [reply acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/SyncingStatusAcceptanceTest.java:76]: Done

**Comment by mehdi-aouadi** [comment]
> LGTM Just when we start with EL offline we are responding with `el_offline` `false` in first 30 seconds until we get error, we could change this behavior but it will mean that response is not correct when el is online on start, so it's a normal trade-off I think

Yep we only know the real EL state after a call. I guess assuming it's available and then updating the status whenever a call fails/succeeds is an acceptable trade-off

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ For information on changes in released versions of Teku, see the [releases page]
 - Introduced [Doppelganger Detection feature](https://docs.teku.consensys.net/en/latest/HowTo/Doppelganger-Detection/) 
 - Changed color of block production duty to cyan, rather than dark blue.
 - Introduced a configuration overview printed during start up.
+- Implemented the new [Syncing API update](https://github.com/ethereum/beacon-APIs/pull/290): Introduced a new field `el_offline` in the `GetSyncingStatusResponse` which is set to `true` when the execution layer is offline.
```
Comment (rolfyone) [inline CHANGELOG.md]: We should be careful how we phrase this, we didnt implement a new api, we added the new optional field to the existing syncing api...

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/node/GetSyncingTest.java [hunk 3]
```diff
@@ -70,9 +81,9 @@ void metadata_shouldHandle500() throws JsonProcessingException {
   void metadata_shouldHandle200() throws JsonProcessingException {
     final String data =
         getResponseStringFromMetadata(
-            handler, SC_OK, new GetSyncing.SyncStatusData(false, false, 10, 0));
+            handler, SC_OK, new GetSyncing.SyncStatusData(false, false, false, 10, 0));
     assertThat(data)
         .isEqualTo(
-            "{\"data\":{\"head_slot\":\"10\",\"sync_distance\":\"0\",\"is_syncing\":false,\"is_optimistic\":false}}");
+            "{\"data\":{\"head_slot\":\"10\",\"sync_distance\":\"0\",\"is_syncing\":false,\"is_optimistic\":false,\"el_offline\":false}}");
   }
```
Comment (rolfyone) [inline data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/node/GetSyncingTest.java]: I'd like to see a test added that shows we can have the field not specified...

File: CHANGELOG.md (hunk 2 above)
Comment (mehdi-aouadi) [reply CHANGELOG.md]: I rephrased it event though It didn't mention a new api but a new API update...
Must be clearer now

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/node/GetSyncingTest.java (hunk 3 above)
Comment (mehdi-aouadi) [reply data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/node/GetSyncingTest.java]: Good catch! I added a test for the `isOptimistic` too...

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**